### DISCUSSION TITLE
How to make my custom module as homepage?

### DISCUSSION THREAD

**Discussion by MRSheikh** [OrchardCMS/OrchardCore#8412] [Q&A] [answered]
hi.
i created orchardcore CMS and create custom module for it. so i want to make my one of my module page as homepage of CMS.
is it possible and how do it?

**Comment by DotCat1985** [comment, marked as answer]
If the home page is a content item, activate `OrchardCore.HomeRoute` module and type the following code on JSON recipe:

```
{
  ...
  "steps": [
    {
      "name": "feature",
      ...
      "enable": [
        ...
        "OrchardCore.HomeRoute",
		...
      ]
    },
	...
    {
      "name": "Settings",
	  ...
      "HomeRoute": {
        "Action": "Display",
        "Controller": "Item",
        "Area": "OrchardCore.Contents",
        "ContentItemId": "CONTENT_ITEM_ID_OF_YOUR_MODULE"
      }
    }
  ]
}

```

Else if the home page is an action written inside your module, activate `OrchardCore.HomeRoute` module and type the following code on JSON recipe:

```
{
  ...
  "steps": [
    {
      "name": "feature",
      ...
      "enable": [
        ...
        "OrchardCore.HomeRoute",
		...
      ]
    },
	...
    {
      "name": "Settings",
	  ...
      "HomeRoute": {
        "Action": "ACTION_NAME",
        "Controller": "CONTROLLER_NAME",
        "Area": "MODULE_NAME",
        "ROUTE_ID": "VALUE"
      }
    }
  ]
}

```
On [Orchard Recipes Documentation](https://docs.orchardcore.net/en/dev/docs/reference/modules/Recipes/#settings-step) you find an example about your requirement.
If you know .NET and inspect the CMS code, `OrchardCore.HomeRoute` module is based on the using of [RouteValueDictionary](https://docs.microsoft.com/it-it/dotnet/api/system.web.routing.routevaluedictionary?view=netframework-4.8) object.

**Reply by MRSheikh** [reply]
thanks a lot. that's work fine

```
"HomeRoute": {
        "Action": "ACTION_NAME",
        "Controller": "CONTROLLER_NAME",
        "Area": "MODULE_NAME",
        "ROUTE_ID": "VALUE"
      }

**Comment by hishamco** [comment]
If your page listed you can set it as "Home" from the admin site
